# Inventory Analysis API Test Script

## Overview
This Python script (`analyze_test.py`) is designed to test the functionality of the Inventory Analysis API. It sends a sample request to the API endpoint `/analyze` with predefined data and prints the response.

## Running the Test Script
1. **Requirements:**
    - Python
    - Requests library (`pip install requests`)

2. **Run the Test Script:**
   - Navigate to the directory containing the test script.
   - Execute the following command in the terminal:
     ```bash
     python analyze_test.py
     ```

3. **Output:**
   - The script will print the HTTP status code and the JSON response received from the API.

## Important Notes
- Ensure that the API (`your_api_file.py`) is running locally before executing the test script.
- Adjust the API endpoint URL and request data in the script as needed.


In [1]:
import requests

# Define the URL for your API endpoint
url = 'http://127.0.0.1:5000/analyze'

# Define the request data in JSON format
data = {
    "startDate": "01/01/2023",
    "endDate": "31/12/2023",
    "branch": "DAN002",
    "itemCode": "SAL009"
}

# Send a POST request to the API endpoint
response = requests.post(url, json=data)

# Print the response from the API
print(response.status_code)
print(response.json())


200
{'branch': 'DAN002', 'itemCode': 'SAL009', 'itemData': {'actualOOSD': 187, 'averageOOSD': 17.0, 'endDate': '31/12/2023', 'startDate': '01/01/2023', 'totalPurchases': 89995, 'totalSales': 84753}}


## Interpretation of Analysis Result

**Interpretation:**

- The analysis for branch 'DAN002' and item code 'SAL009' during the period from 01/01/2023 to 31/12/2023 indicates the following:
  - Actual Out Of Stock Days (actualOOSD): 187 days, which means the product was out of stock for a total of 187 days during this period.
  - Average Out Of Stock Days (averageOOSD): 17.0 days, indicating that, on average, the product was out of stock for approximately 17 days during each restocking cycle.
  - Total purchases during this period: 89,995 items, reflecting the quantity of 'SAL009' purchased.
  - Total sales during this period: 84,753 items, representing the quantity of 'SAL009' sold.

**Significance:**

- The analysis highlights that for branch 'DAN002' and item code 'SAL009,' there were significant instances of running out of stock, totaling 187 days.
- The average out-of-stock period of 17 days suggests that there were frequent occurrences of the product being unavailable for purchase.
- To improve customer satisfaction and avoid lost sales opportunities, it's crucial to manage stock levels effectively, potentially by increasing purchase quantities or optimizing restocking schedules.
